stdlib/src/hashlib.rs

```diff
@@ -260,8 +265,24 @@ mod hashlib {
         init(PyHasher::new("blake2s", HashWrapper::blake2s()), args.data)
     }
 
-    trait ThreadSafeDynDigest: DynDigest + Sync + Send {}
-    impl<T> ThreadSafeDynDigest for T where T: DynDigest + Sync + Send {}
+    trait ThreadSafeDynDigest: DynDigest + Sync + Send {
+        fn box_clone(&self) -> Box<dyn ThreadSafeDynDigest>;
+    }
+
+    macro_rules! impl_thread_safe_dyn_digest {
+        ({$($t:ty),+}) => {
+            $(
+                impl ThreadSafeDynDigest for $t {
+                    fn box_clone(&self) -> Box<dyn ThreadSafeDynDigest> {
+                        Box::new(self.clone())
+                    }
+                }
+            )+
+        };
+    }
+
+    impl_thread_safe_dyn_digest!({Md5, Sha1, Sha224, Sha256, Sha384, Sha512,
+        Sha3_224, Sha3_256, Sha3_384, Sha3_512, Blake2b512, Blake2s256});
```

Comment:
maybe you wanted to use https://crates.io/crates/dyn-clone

Reply:
Yes indeed!
I updated the code to use the dyn-clone crate instead!